docs/architecture_overview.md

```diff
@@ -0,0 +1,130 @@
+### Projects
+While we can think of a `TaskRun` as a component of a `Task`, it's also useful to think of a `Task` as a component of a `Project`. The goal behind a `Project` is to collect analysis information about multiple tasks that aren't similar enough to be combined, but are part of some larger objective. That is to say, you can't create an interface that allows the subtasks of a project to be completed together, but the subtasks are still related.
+
+Projects can be useful to aggregate information about those subtasks in one place, such that they may be utilized in positive ways. Here is where you may be able to find the total budget of a larger group of work, or determine a set of workers with high approval rates for this work. You could also use a `Project` more for archival reasons, marking where you can free up some disk space once the work is final.
+
+> **Note**: The `Project` class is largely un-implemented at this point, and its convenience methods are likely to come in future releases.
+
+## Assignments
+Where the `Task` classes are used to define big pieces of work that need to be done, the `Assignment` and `Unit` classes are used to drill down on the work that you'll actually be showing to a worker. Here, a `TaskRun` is comprised of one or more `Assignment`'s, and an `Assignment` is comprised of one or more `Unit`'s. The goal of this separation is to make different types of assignment-unit breakdowns valid, such as cooperation or consensus.
+
+### Assignment
+An `Assignment` can be thought of the minimal subgoal of a given `Task`. If your `Task` is to *Annotate [these] images*, one such subgoal could be to *Annotate [this] image*. It may be appropriate to batch these, such that the assignment is actually to *Annotate [these 5] images*, but the goal here is to ensure that an `Assignment` represents a reasonable amount of work to accomplish, but small enough to be reviewed as a whole.
+
+In some cases, an `Assignment` may relate to something that needs to be done by multiple workers at once, such as a cooperative task. In other cases, an `Assignment` may relate to wanting multiple different workers to attempt the same task in order to get consensus. We refer to this as a task being *concurrent* or not. In either case, however, Mephisto will prevent any given `Worker` from participating in any `Assignment` more than once.
+### Unit
+The `Unit` is the smallest bit of work that is part of a task, in that it comprises a single worker's contribution to that task. It has the required information for initializing a specific instance of work to the worker, and is responsible for having up-to-date information about the real-life status of that work.
+
+Some `Assignment`'s may have just one `Unit`, should the task be something that can be completable by just one worker and only needs completing once. Some `Assignment`'s, like a conversation, may be comprised of two (or more) `Unit`'s, each representing one worker's place in the conversation.
```

Comment:
How about in a consensus situation? Is it similar in that a single `Assignment` will have multiple `Unit`s assigned concurrently?